code/React/BootstrapComponent.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace SilverStripe\Admin\React;
+
+use SilverStripe\Core\Convert;
+
+trait BootstrapComponent
```

Comment:
This trait contains most of the logic that needs to be added to a ViewableData object to be rendered as a React component.

Reply:
The trait is coupled to `ViewableData`, which to me implies that this should be a subclass, not a trait. What are the tradeoffs for asking devs to subclass `ReactComponent` or something?